### PR TITLE
Scripts/Spells: implement warrior spell Shield Wall

### PR DESCRIPTION
**Pull request by HannibalRoG**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Implement spell for warrior class Wall shield.

**Target branch(es):** 3.3.5/master

- [x] master

**Issues addressed:** Closes #  (insert issue tracker number)
Don´t find any

**Tests performed:** (Does it build, tested in-game, etc.)
Compiled and tested in game

**Known issues and TODO list:** (add/remove lines as needed)

- [ ]  I dont think the visual are correct, i cant find any video from retail.
- [ ]  I dont know if visual for Horde and Alliance exist in Legion.

### CODE REVIEW

**Comment by DoctorKraft** [comment]
I don't understand, the visual on spell are correct, don't need others visual. I have check on retail

**Comment by HannibalRoG** [comment]
With shield and without it when you cast the spell (can do that without a shield equiped?) the visuals are differents, i dont know if this is correct

**Comment by DoctorKraft** [comment]
I have check for ally (no shield) ally (with shield) same for horde, it's same visual for all

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 1]
```diff
@@ -866,6 +870,64 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+        }
+
+        void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetCaster())
+                return;
+
+            if (Player* _player = GetCaster()->ToPlayer())
+            {
+                if (_player->HasAura(SPELL_WARRIOR_SHIELD_OF_WALL_NOSHIELD))
```
Comment (Temppus) [inline src/server/scripts/Spells/spell_warrior.cpp]: Are those HasAura checks necessary ?

**Comment by wonder2011** [comment]
@DoctorKraft can you post a photo here of how the spell is on retail?. You say that Ali, Horde and with Shield and no shield is the same, so please upload a photo to see how is this and make the final implementation. @Temppus i will tell @HannibalRoG about your comment to help them fix this.

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 2]
```diff
@@ -885,17 +882,9 @@ class spell_warr_shield_wall : public SpellScriptLoader
             if (!GetCaster())
                 return;
 
-            if (Player* _player = GetCaster()->ToPlayer())
+            if (Player* player = GetCaster()->ToPlayer())
```
Comment (Temppus) [inline src/server/scripts/Spells/spell_warrior.cpp]: Why cast to player ?

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 3]
```diff
@@ -904,25 +893,20 @@ class spell_warr_shield_wall : public SpellScriptLoader
...
```
Comment (Temppus) [inline src/server/scripts/Spells/spell_warrior.cpp]: Why cast to player ?

**Comment by DoctorKraft** [comment]
http://i.imgur.com/p76ByaM.jpg the visual base are already good, don't need this script. Maybe you confuse with spell reflection ?

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 4]
```diff
@@ -866,6 +867,51 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+        PrepareAuraScript(spell_warr_shield_wall_AuraScript);
+
+        void OnApply(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetCaster())
+                return;
+
+            if (Player* player = GetCaster()->ToPlayer())
+            {
+                player->AddAura(SPELL_WARRIOR_SHIELD_OF_WALL, player);
```
Comment (Keader) [inline src/server/scripts/Spells/spell_warrior.cpp]: Why add aura?

File: src/server/scripts/Spells/spell_warrior.cpp [hunk 5]
```diff
@@ -866,6 +867,51 @@ class spell_warr_second_wind_trigger : public SpellScriptLoader
...
+        }
+
+        void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+        {
+            if (!GetCaster())
+                return;
+
+            if (Player* player = GetCaster()->ToPlayer())
+            {
+                player->RemoveAura(SPELL_WARRIOR_SHIELD_OF_WALL);
```
Comment (Keader) [inline src/server/scripts/Spells/spell_warrior.cpp]: RemoveAurasDueToSpell(SPELL_WARRIOR_SHIELD_OF_WALL);
And i cant see any reason to cast GetCaster() to player

**Comment by wonder2011** [comment]
Then, @DoctorKraft you are saying that spells like this don't need scripting in Core ?. Why?. I thing i miss something in BD that make this spell work fully witouth script. If this is the reason, please let me know to learn.

**Comment by DoctorKraft** [comment]
http://www.wowhead.com/spell=871/shield-wall you speak about this spell ?

**Comment by wonder2011** [comment]
@DoctorKraft yes, the spell Shield Wall - (ID 871), but the question is about how this system works in TC.

**Comment by DoctorKraft** [comment]
WTF dude this spell need nothing, you have already visual and effect are already scripted in core `SPELL_AURA_MOD_DAMAGE_PERCENT_TAKEN = - 40` Look this http://i.imgur.com/p76ByaM.jpg

**Comment by wonder2011** [comment]
@DoctorKraft rKraft ok thanks for clarify this.

File: src/server/scripts/Spells/spell_warrior.cpp (hunk 5 above)
Comment (tkrokli) [inline src/server/scripts/Spells/spell_warrior.cpp]: @HannibalRoG : Please rename this enum to `SPELL_WARRIOR_SHIELD_WALL`
The spell is named "Shield Wall", not Shield Of Wall.

@Keader : you may want to repost your comments after it has been changed.

**Pull request closed without merging**